src/main/java/ie/gmit/softwareeng/labexam/Playlist.java

```diff
@@ -1,71 +1,105 @@
 package ie.gmit.softwareeng.labexam;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 public class Playlist {
+
+    private String name;
+
+    private List<Song> songs = new ArrayList<>();
+
+    private Song currentSong;
```

Comment:
You could have stored the index of the song instead of referencing the song object here. It would have saved you having to call songs.indexOf in a few places. This works perfectly though